docs/guides/unified-provider.md

```diff
@@ -0,0 +1,121 @@
+
+### Block
+```hcl
+resource "workspace_level_resource" "this" {
+    provider_config {
+        workspace_id = "12345"
+    }
+    ...
+}
+```
+
+### Attribute
+```hcl
+resource "workspace_level_resource" "this" {
+    provider_config = {
+        workspace_id = "12345"
+    }
+    ...
+}
+```
+
+**Note:** This feature is being rolled out incrementally. Some resources do not yet support the unified provider. Please check the resource-specific documentation to see if the `provider_config` attribute or block is available.
+
+## Migrating to Unified Provider
+
+If you are currently managing both workspace-level and account-level resources and data sources, you likely have multiple provider configurations that you specify for each resource using aliases. For example:
+```hcl
+// Define an account provider with alias
+provider "databricks" {
+    alias         = "account"
+    host          = var.account_host
+    account_id    = var.account_id
+    client_id     = var.account_client_id
+    client_secret = var.account_client_secret
+}
+
+// Use the account provider for account-level resources
+resource "databricks_account_federation_policy" "this" {
+    provider = databricks.account
+    policy_id = "my-policy"
+    oidc_policy = {
+        issuer = "https://myidp.example.com"
+        subject_claim = "sub"
+    }
+}
+
+// Define a workspace provider with alias
+provider "databricks" {
+    alias           = "workspace"
+    host            = var.workspace_host
+    client_id       = var.workspace_client_id
+    client_secret   = var.workspace_client_secret
+}
+
+// Use the workspace provider for workspace-level resources
+resource "databricks_workspace_level_resource" "this" {
+    provider = databricks.workspace
+    name = "resource name"
+}
+```
+
+To migrate to the unified provider, you can remove the workspace-level provider and specify the `workspace_id` in the `provider_config` attribute or block instead. For example:
+
+```hcl
+// Define an account provider
+provider "databricks" {
+    host          = var.account_host
+    account_id    = var.account_id
+    client_id     = var.account_client_id
+    client_secret = var.account_client_secret
+}
+
+// Create an account-level resource
+resource "databricks_account_federation_policy" "this" {
+    policy_id = "my-policy"
+    oidc_policy = {
+        issuer = "https://myidp.example.com"
+        subject_claim = "sub"
+    }
+}
+
+// Create a workspace under the account
+resource "databricks_mws_workspaces" "this" {
+    account_id   = var.account_id
+    aws_region   = "us-east-1"
+    compute_mode = "SERVERLESS"
+}
+
+// Create a workspace-level resource using provider_config
+resource "databricks_workspace_level_resource" "this" {
+    provider_config {
+        workspace_id = databricks_mws_workspaces.this.workspace_id
+    }
+    name = "resource name"
+}
+```
+
```

Comment:
Suggestion: include an FAQ, and describe the error that happens when users don't specify workspace_id (or try to migrate in one step rather than two) & what happens when workspace ID doesn't match.